pkg/resultx/resultx.go

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package resultx
+
+import (
+	"github.com/arana-db/arana/pkg/proto"
+)
+
+var (
+	_ proto.Result = (*noopResult)(nil)
+	_ proto.Result = (*halfResult)(nil)
+	_ proto.Result = (*fullResult)(nil)
+	_ proto.Result = (*dsOnlyResult)(nil)
```

Comment:
What's the difference of these result? `noopResult`, `halfResult` ...

Reply:
the `Result` should be designed low-memory cost:
1. noop => empty result, cost zero mem because of empty struct
2. half => contains `rowsAffected` and  `lastInsertId` only, normally for the operation includes insert/update/delete..., only cost 16 byte
3. full => fully result, contains all elements.